src/util_libc.rs

```diff
@@ -5,9 +5,30 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+extern crate std;
 
 use crate::util::LazyUsize;
+use crate::Error;
 use core::ptr::NonNull;
+use std::io;
+
+pub fn fill_exact(mut buf: &mut [u8], f: impl Fn(&mut [u8]) -> libc::ssize_t) -> Result<(), Error> {
+    while !buf.is_empty() {
+        let res = f(buf);
+        if res < 0 {
+            let err = io::Error::last_os_error();
```

Comment:
Seems odd to me to assume that calling `last_os_error` is appropriate here given the function name. At least this should be mentioned by function documentation.

Reply:
During removal of `std` dependency it will have to be replaced with an explicit `errno`, so I think it's fine to keep it as-is for the time being.

Reply:
So I changed the function to be named `sys_fill_exact` and added a comment to explain the expected semantics of the `sys_fill` function. @newpavlov is right that this will be replaced with an explicit call to an errno function in #54